projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-jitm-v2.php

```diff
@@ -0,0 +1,184 @@
+<?php
+/**
+ * REST API endpoint for retrieving JITMs from the WPCOM API via the Jetpack JITM class
+ * infrastructure.
+ *
+ * Replaces projects/packages/jitm/src/class-rest-api-endpoints.php.
+ *
+ * Available on:
+ * - Simple - via Dotcom Public API (https://public-api.wordpress.com/wpcom/v2/sites/{site_id}/jitm-v2).
+ * - WoA and Jetpack connected sites - via local site REST API (https://myjetpackconnectedsite.com/wp-json/wpcom/v2/jitm-v2)
+ *
+ * Utilises Jetpack classes to orchestrate the request and response handling.
+ * All JITM configuration happens on the Dotcom Simple codebase.
+ *
+ * @package automattic/jetpack
+ */
+
+use Automattic\Jetpack\Connection\REST_Connector;
+
+/**
+ * Class WPCOM_REST_API_V2_Endpoint_JITM_V2
+ */
+class WPCOM_REST_API_V2_Endpoint_JITM_V2 extends WP_REST_Controller {
+
+	/**
+	 * Namespace prefix.
+	 *
+	 * @var string
+	 */
+	public $namespace = 'wpcom/v2';
+
+	/**
+	 * Endpoint base route.
+	 *
+	 * @var string
+	 */
+	public $rest_base = 'jitm-v2';
+
+	/**
+	 * WPCOM_REST_API_V2_Endpoint_JITM_V2 constructor.
+	 */
+	public function __construct() {
+		add_action( 'rest_api_init', array( $this, 'register_routes' ) );
+	}
+
+	/**
+	 * Register JITM V2 routes.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			$this->rest_base,
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_item' ),
+					'permission_callback' => '__return_true',
```

Comment:
Should we limit this to logged in users?

Reply:
I'm mirroring the check on the existing endpoint but I think it should be ok to change the permission to at least be logged in.